protobuf-maven-plugin/src/main/java/io/github/ascopes/protobufmavenplugin/OptionalProtocPlugin.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (C) 2023 - 2024, Ashley Scopes.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.github.ascopes.protobufmavenplugin;
+
+import org.immutables.value.Value.Default;
+
+/**
+ * Protoc plugin base that can be optionally ignored if it cannot be resolved.
+ *
+ * <p>By default, plugins are <strong>NOT</strong> optional.
+ *
+ * @author Ashley Scopes, Anthony Alayo
```

Comment:
@anthonyalayo new class, but you implemented the original change so I've attributed you here if that is okay with you!